File: app/src/main/java/com/kidozh/discuzhub/results/FavoriteForumResult.java
package com.kidozh.discuzhub.results;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.kidozh.discuzhub.entities.FavoriteForum;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class FavoriteForumResult extends BaseResult {
    @JsonProperty("Variables")
    public FavoriteForumVariable FavoriteForumVariable;

    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class FavoriteForumVariable extends VariableResults{
        @JsonProperty("list")
        public List<FavoriteForum> FavoriteForumList;
        @JsonProperty("perpage")
        @JsonFormat(shape = JsonFormat.Shape.STRING)
        public int perpage;
        @JsonProperty("count")
        @JsonFormat(shape = JsonFormat.Shape.STRING)
        public int count;

    }





    public boolean isError(){
        return this.message != null;
    }

}

File: app/src/main/java/com/kidozh/discuzhub/results/FavoriteForumResult.kt
package com.kidozh.discuzhub.results

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.kidozh.discuzhub.entities.FavoriteForum

@JsonIgnoreProperties(ignoreUnknown = true)
class FavoriteForumResult : BaseResult() {
    @JsonProperty("Variables")
    var favoriteForumVariable: FavoriteForumVariable = FavoriteForumVariable()

    @JsonIgnoreProperties(ignoreUnknown = true)
    class FavoriteForumVariable : VariableResults() {
        @JsonProperty("list")
        var FavoriteForumList: List<FavoriteForum> = ArrayList()

        @JsonProperty("perpage")
        @JsonFormat(shape = JsonFormat.Shape.STRING)
        var perpage = 0

        @JsonProperty("count")
        @JsonFormat(shape = JsonFormat.Shape.STRING)
        var count = 0
    }

    override fun isError(): Boolean {
        return message != null
    }
}